x-pack/plugins/apm/public/components/app/service_overview/service_overview_errors_table/index.tsx

```diff
@@ -96,46 +66,23 @@ export function ServiceOverviewErrorsTable({ serviceName }: Props) {
             },
           },
         }
-      ).then((response) => {
-        const currentPageErrorGroups = orderBy(
-          response.errorGroups,
-          field,
-          direction
-        ).slice(pageIndex * PAGE_SIZE, (pageIndex + 1) * PAGE_SIZE);
-
-        return {
-          // Everytime the main statistics is refetched, updates the requestId making the comparison API to be refetched.
-          requestId: uuid(),
-          items: currentPageErrorGroups,
-          totalItems: response.errorGroups.length,
-        };
-      });
+      );
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      environment,
-      kuery,
-      start,
-      end,
-      serviceName,
-      pageIndex,
-      direction,
-      field,
-      // not used, but needed to trigger an update when offset is changed either manually by user or when time range is changed
-      offset,
-      // not used, but needed to trigger an update when comparison feature is disabled/enabled by user
-      comparisonEnabled,
-    ]
+    [environment, kuery, start, end, serviceName]
   );
 
-  const { requestId, items, totalItems } = data;
+  const { requestId, tableItems, onTableChange, tablePagination, tableSort } =
+    useTableSortAndPagination(
+      { items: data.errorGroups, sorting: { sort: { field: 'occurrences' } } },
+      [offset, comparisonEnabled]
```

Comment:
Outside the scope of this but it would be great to have these two options grouped together:

```suggestion
      [comparisonOffset, comparisonEnabled]
```

or

```suggestion
      [comparison.offset, comparison.enabled]
```